config-model/src/main/java/com/yahoo/vespa/model/builder/xml/dom/NodesSpecification.java

```diff
@@ -344,9 +363,28 @@ private static IllegalArgumentException clusterReferenceNotFoundException(String
         return dockerImageFromElement == null ? dockerImage : Optional.of(dockerImageFromElement);
     }
 
+    /** Parses a value ("value") or value range ("[min-value, max-value]") */
+    private static <T> Pair<T, T> toRange(String s, T defaultValue, Function<String, T> valueParser) {
+        try {
+            if (s == null) return new Pair<>(defaultValue, defaultValue);
+            s = s.trim();
+            if (s.startsWith("[") && s.endsWith("]")) {
+                String[] numbers = s.substring(1, s.length() - 1).split(",");
+                if (numbers.length != 2) throw new IllegalArgumentException();
+                return new Pair<>(valueParser.apply(numbers[0].trim()), valueParser.apply(numbers[1].trim()));
+            } else {
+                return new Pair<>(valueParser.apply(s), valueParser.apply(s));
+            }
+        }
+        catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Expected a number or range on the form [min, max], but got '" + s + "'");
```

Comment:
Consider adding the message from `e`, useful when `valueParser` is `parseGbAmount` and users specify invalid unit.